content/blog/gmailr-2-0-0/index.Rmd

```diff
@@ -0,0 +1,183 @@
+  rappdirs::user_data_dir("gmailr") |>
+    list.files()
+  #> [1] "client_secret_xxx-yyy.apps.googleusercontent.com.json"
+  
+  gm_default_oauth_client()
+  #> [1] "/Users/jenny/Library/Application Support/gmailr/client_secret_xxx-yyy.apps.googleusercontent.com.json"
+  ```
+  
+  `gm_default_oauth_client()` is the new function that implements this new
+  feature as well as pre-existing support for providing this path via an
+  environment variable.
+* If the OAuth client is configured for auto-discovery, it is no longer
+  necessary to call `gm_auth_configure()` explicitly. That is taken care of
+  internally, inside `gm_auth()`.
+  
+  ```{r eval = FALSE}
+  > library(gmailr)
+  > 
+  > # 😲 OMG no more need to call gm_auth_configure() here! 🎉
+  > 
+  > gm_threads()
+  The gmailr package is requesting access to your Google account.
+  Enter '1' to start a new auth process or select a pre-authorized account.
+  1: Send me to the browser for a new auth process.
+  2: jenny@posit.co
+  Selection: 2
+  ```
+  
+* Conversely, if you happen to be providing an explicit user or service account
+  token, `gm_auth(token =)` and `gm_auth(path =, subject =)`[^1] no longer error if
+  the OAuth client is not configured.
+  
+[^1]: The `subject` argument of `gm_auth()` is also new and facilitates the use of a service account to impersonate a user.
+  
+## Auth in a deployed or other non-interactive setting
+
+The Gmail API is primarily intended for use on behalf of a regular Google user account.
+The gmailr package is designed to guide an interactive R user through a process in which they authenticate themselves to Google and authorize Gmail activities initiated from R.
+This is sometimes referred to as the "OAuth dance".[^2]
+
+[^2]: The full OAuth dance is not necessary in subsequent R sessions, though, by default gmailr is very conservative and asks for permission to use and refresh an existing token. This is, of course, configurable.
+
+But what about settings where there is no interactive user sitting around to do this dance, i.e. when gmailr-using code is deployed to a remote server or otherwise runs unattended?
+For most Google APIs, the standard advice is "use a service account".
+But the Gmail API is special.
+To use a service account with the Gmail API basically requires that the service account has been delegated domain-wide authority.
+This is tricky for at least two reasons.
+First, this is only possible within a Google Workspace.
+It's not available to personal Google accounts.
+Second, most Google Workspace admins will refuse to do this, for security reasons.
+
+Therefore, if you want to deploy a data product that uses gmailr, it's extremely likely that you really do need to use a user token.
+This workflow has gotten dramatically easier in gmailr v2.0.0:
+
+* [Deploy a token](https://gmailr.r-lib.org/articles/deploy-a-token.html) is a new article describing how to capture a token interactively, then use it later, non-interactively.
```

Comment:
This link 404s but won't once the release is published. I didn't bother with the hack in this case because this is a real vignette and I can refer to it with `vignette()` syntax inside the package.